### PR TITLE
[casr-afl][casr-libfuzzer] Add progress printing

### PR DESCRIPTION
**Pull request by hkctkuy**

### CODE REVIEW

File: casr/src/bin/casr-afl.rs [hunk 1]
```diff
@@ -378,25 +386,32 @@ fn summarize_results(
             .filter(|e| !Path::new(format!("{}.gdb.casrep", e.display()).as_str()).exists())
             .collect();
         let num_of_threads = jobs.min(crashes.len());
-        if num_of_threads > 0 {
+        if num_of_threads > 1 {
```
Comment (SweetVishnya) [inline casr/src/bin/casr-afl.rs:389]: jobs.min(crashes.len() + 1)

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: casr/src/bin/casr-afl.rs [hunk 2]
```diff
@@ -263,20 +264,27 @@ fn main() -> Result<()> {
...
-            .par_iter()
-            .try_for_each(|(_, crash)| crash.run_casr(output_dir.as_path(), timeout))
-    })?;
+    info!("Using {} threads", num_of_threads - 1);
+    let counter = RwLock::new(0_usize);
+    let total = crashes.len();
+    custom_pool.join(
+        || {
+            let _ = crashes.par_iter().try_for_each(|(_, crash)| {
+                crash.run_casr(output_dir.as_path(), timeout)?;
```
Comment (SweetVishnya) [inline casr/src/bin/casr-afl.rs:281]: Program may hang on error here and in casr-libfuzzer

Comment (hkctkuy) [reply casr/src/bin/casr-afl.rs]: Done

File: casr/src/bin/casr-afl.rs [hunk 3]
```diff
@@ -275,16 +275,22 @@ fn main() -> Result<()> {
...
-                Ok::<(), anyhow::Error>(())
-            });
-        },
-        || util::log_progress(&counter, total),
-    );
+    custom_pool
+        .join(
+            || {
+                crashes.par_iter().try_for_each(|(_, crash)| {
+                    let Ok(_) = crash.run_casr(output_dir.as_path(), timeout) else {
```
Comment (SweetVishnya) [inline casr/src/bin/casr-afl.rs:282]: if let Err(e) = ... {
   bail!( Print error message here

Comment (SweetVishnya) [reply casr/src/bin/casr-afl.rs:282]: the same for casr-libfuzzer

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: casr/src/bin/casr-afl.rs (hunk 3 above)
Comment (hkctkuy) [reply casr/src/bin/casr-afl.rs]: Done

File: casr/src/bin/casr-libfuzzer.rs [hunk 4]
```diff
@@ -180,44 +180,44 @@ fn main() -> Result<()> {
...
-                ]);
-                if !atheris_asan_lib.is_empty() {
-                    casr_cmd.arg("python3");
-                    casr_cmd.env("LD_PRELOAD", &atheris_asan_lib);
-                }
-                casr_cmd.args(argv.clone());
-                casr_cmd.arg(crash);
-                debug!("{:?}", casr_cmd);
+                if let Err(e) = crashes.par_iter().try_for_each(|(crash, fname)| {
+                    let mut casr_cmd = Command::new(tool);
```
Comment (SweetVishnya) [inline casr/src/bin/casr-libfuzzer.rs:184]: This code is from Ok case

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**